index.html

```diff
@@ -257,12 +276,21 @@
         <a title="user agent">User agents</a> MUST NOT provide Push API access to <a title=
         "webapp">webapps</a> without the <a>express permission</a> of the user. <a title=
         "user agent">User agents</a> MUST acquire consent for permission through a user interface
-        for each call to the <code>subscribe()</code> method, unless a prearranged trust
-        relationship applies.
+        for each call to the <code>subscribe()</code> method, unless a previous permission grant
+        has been persisted, or a prearranged trust relationship applies. Permissions that are
+        preserved beyond the current browsing session MUST be revocable.
       </p>
       <p>
-        <a title="user agent">User agents</a> MAY support prearranged trust relationships that do
-        not require such per-request user interfaces.
+        When a permission is revoked, all <a title="push subscription">push subscriptions</a>
+        created with that permission MUST be terminated.
+      </p>
+      <p>
+        When a <a>service worker registration</a> is unregistered, any associated <a>push
+        subscription</a> MUST be terminated.
+      </p>
+      <p>
+        The <a>subscription id</a> of a terminated <a>push subscription</a> MUST never be reused
+        for a new <a>push subscription</a>.
```

Comment:
Perhaps concisely clarify *why* this is the case? (Persistent identifiers.)

I imagine that an implementer reading the specification will initially consider the semantics of the push service that will service their implementation. If that service generates a subscription Id based on some kind of hardware Id, it's not unrealistic to expect naive re-use. Hence why giving some context may be useful.